python/TestHarness/testers/Tester.py

```diff
@@ -71,21 +71,52 @@ def __init__(self, name, params):
 
     # Set the status message
     if self.specs['check_input']:
-      self.status_message = 'SYNTAX PASS'
+      self.success_message = 'SYNTAX PASS'
     else:
-      self.status_message = self.specs['status_message']
+      self.success_message = self.specs['success_message']
+
+    # Initialize the status bucket
+    self.status = {}
+
+    # Initialize the 'reason' a test failed. Empty means it passed.
+    self.status_message = ''
 
     # Set up common paramaters
     self.should_execute = self.specs['should_execute']
     self.check_input = self.specs['check_input']
+    self.success_bucket = 'PASS'
 
   # Method to return the input file if applicable to this Tester
   def getInputFile(self):
     return None
 
+  # Method to return the successful message printed to stdout
+  def getSuccessMessage(self):
+    return self.success_message
+
+  # Method to return status text (exodiff, crash, skipped because x, y and z etc)
   def getStatusMessage(self):
     return self.status_message
 
+  # Method to return enumerated status (PASS, FAIL, SKIP, DIFF)
+  def getStatus(self):
+    if self.getStatusMessage() in self.status:
+      return self.status[self.getStatusMessage()]
+    return None
+
+  # Method to set the bucket status
+  def setStatus(self, reason, bucket):
+    self.status_message = reason
+    self.status[reason] = bucket
+    return self.getStatus()
+
+  # Method to check for successfull test
+  def didPASS(self):
+    if self.getStatusMessage() == '':
+      self.setStatus(self.getSuccessMessage(), self.success_bucket)
+      return True
+    return False
```

Comment:
OK, I think I understand what you are trying to do here. It's possible that the Tester never had a status set all the way until the call to this point. In that case you actually set it. The problem is that there's an assumption here that nobody will ever set a "successful" status early. This method also doesn't work if it's accidentally called twice in a row!

Perhaps a better design would be to start with an OK status (assume that a newly created tester is actually OK!). This method would simply return the status.